### ISSUE TITLE
[PNI] Create new streamfield block for primary related articles on article pages

### ISSUE THREAD

**Issue by nancyt1** [MozillaFoundation/foundation.mozilla.org#9874]
## Description

Follow up from #9664
[Figma Designs (includes desktop, mobile, and tablet)](https://www.figma.com/file/GwdoUiY4fzVyeLfXsAJGPq/*Privacy-Not-Included-Website?node-id=3406%3A38682&t=UCOqvUJixeY29Pqv-1)

The idea is to create a stream field block that allows editors to define a location in the article body text to show links to related articles with the indented look displayed below. 

We are using a stream field block to ensure proper alignment of the paragraph and the link element and avoid the number of special cases that can occur that would break the layout (see #9664 for details). 

The new block should have two sections: 
1. a paragraph richtext block that looks like the rest of the body paragraphs, 
2. a list block of page chooser blocks to link to other article pages.

## Acceptance Criteria

- [ ] This would be a block that lets users input only rich text and max 3 links to other article pages. 
- [ ] Related articles section should contain a text link CTA to "All Articles"
- [ ] Text should be able to wrap around the Related Articles section
- [ ] "Related Articles" section itself should take up about the width of the right 4 columns on desktop and top align with text
- [ ] On mobile, the text would stack first then links
- [ ] Max 1 block can be used per page
- [ ] Block should have the attribute `data-gtm="related-articles-block"`
- [ ] Would be helpful to add help text that encourages editors to preview before publishing

## Dev notes

New block type:

* Related articles block (max 1 per page)
  * Paragraph (richtext) block 
  * List block (max 3 items)
    * Page chooser block (limited to other article pages)
    * ...

We need a way to retrieve the content article index page when rendering the block (for the "all articles" link). I think it would be cumbersome to pass that in from the page context. I am also not sure that blocks have access to the page they belong to. Maybe a custom template tag that retrieves the content index page based on the current request. This template tag should definitely cache its result to make it efficient to reuse. 

--- 

![Screen Shot 2022-12-20 at 3.46.00 PM.png](https://images.zenhubusercontent.com/5fb2aa620446ae2c8ea94417/164a6135-d9f6-41e7-93c9-e4e4c9a3a5fa)

**Comment by nancyt1** [comment]
@mmmavis Here's the new ticket!

**Comment by mmmavis** [comment]
@nancyt1 Thanks for creating the ticket. Could you clarify

> This would be a block that lets users input only rich text and max 3 links to other article pages.

On CMS, there's already a "related articles" section outside of `body`. What do we wanna do with it? Are we removing it to avoid confusion?

<img width="289" alt="image" src="https://user-images.githubusercontent.com/2896608/208969666-24b974eb-7a4c-4b07-9674-48a8dc10ba24.png">

--- 

@tbrlpld , just wanna check with you that this is possible to implement and that we can migrate the existing related article links to the new format in the backend. (I assume some articles page already have related links associated with them).

**Comment by nancyt1** [comment]
@mmmavis I was hoping we can move that section to the bottom of the page after "Body" like on the mofo blog. Maybe we can call it "Additional Related Articles" since we wanted 2 different sections for related articles and we're currently testing which one gets more interaction. cc @beccaklam

**Comment by tbrlpld** [comment]
The existing "related articles" is the source for both output locations, primary and secondary related links. 

An automatic transfer is theoretically possible, but hard in this case, because we don't know which paragraph text should be shown next to it. That was what I understood we wanted as a solution, a new stream field block with a paragraph field for the content and a list block of page chooser blocks for the links.

**Comment by tbrlpld** [comment]
🤔 Looks like a bunch of duplicates though:

```
>>> for ar in BuyersGuideArticlePageRelatedArticleRelation.objects.all():
...     print(ar)
... 
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
DE What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
What We Talk About When We Talk About Fitness Trackers -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To?
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
Happy birthday, Amazon. You’re old enough to know better -> In Post Roe v. Wade Era, Mozilla Labels 18 of 25 Popular Period and Pregnancy Tracking Tech With *Privacy Not Included Warning
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
DE What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
Happy birthday, Amazon. You’re old enough to know better -> Calm VS Headspace: Which Is Best For Your Privacy & Security?
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
What We Talk About When We Talk About Fitness Trackers -> How to secure your Ring security camera (because Amazon won’t do it for you)
DE What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
DE What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
DE What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
DE What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
What We Talk About When We Talk About Fitness Trackers -> Mental health apps for teens — what sorts of privacy protections are there?
```

**Comment by tbrlpld** [comment]
Most of them of course come from the localization. 

```
>>> for ar in BuyersGuideArticlePageRelatedArticleRelation.objects.all():
...     print(ar, ar.page.locale.language_code)
... 
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security en
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption pl
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security sw
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security nl
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption fy-NL
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security es
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security fr
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption pt-BR
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security pt-BR
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security fy-NL
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption fr
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption en
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security de
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption es
Getting private? Mental health and prayer apps rush to revise privacy policies -> Top Mental Health and Prayer Apps Fail Spectacularly at Privacy, Security pl
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption de
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption sw
What Can You Tell me About Zoom? -> Zoom Makes the Right Call on Encryption nl
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? pl
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) en
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) nl
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) fr
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) fy-NL
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) de
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) pl
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) pt-BR
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) es
What Can You Tell me About Zoom? -> Tips to Make Your Zoom Gatherings More Private (Updated) sw
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? en
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? nl
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? fr
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? fy-NL
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? de
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? pt-BR
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? es
Getting private? Mental health and prayer apps rush to revise privacy policies -> When My Employer Provides My Mental Health App, How Much Data Do They Have Access To? sw
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? fr
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? en
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? pt-BR
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? nl
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? sw
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? es
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? de
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? fy-NL
Getting private? Mental health and prayer apps rush to revise privacy policies -> Mental health apps for teens — what sorts of privacy protections are there? pl
```

**Comment by tbrlpld** [comment]
So we really only have 5 related articles. We can still keep these as well. 

Previously the idea was to show the first 3 as the primary related articles at the top and the latter 3 as secondary related articles in the bottom. The number of related articles is limited to 6. 

We can reduce that limit to 3 and remove the concept of primary/secondary and only show the  related articles at the bottom (we probably want to relocate the button in the admin also to the bottom of the form so it makes more sense for editors).

Additionally, we create that new block that allows editors to add additional related articles in the body of the page.

**Comment by beccaklam** [comment]
Hi @mmmavis @nancyt1 @tbrlpld! I had a discussion with Kristina about this ticket (mainly because I went to input related articles but then realized this work still needs to happen first) and we think it'd be best if we could break this into smaller chunks, for example:
- Remove the "primary" related articles styling for now since it's causing the most problems
- Have the related articles section in the cms only populate the "secondary" related articles (I agree with Tibor about moving the order the related articles section shows up in the cms). This would unblock us from getting related articles live on the PNI site (currently we're holding off bc of the styling issues) Note: The secondary related articles also needs some spacing adjustments (right now it's a bit too tight under the newsletter signup box)
- Create the new related articles block that can be inserted into the body of the page (what this ticket was originally for)

I'll set up a quick meeting for all of us next week just so we can agree on next steps and then create any outstanding tickets that are needed. Let me know if that sounds good!

**Comment by tbrlpld** [comment]
@nancyt1 I have made some edits to the ticket, can you check this still sounds good to you?

**Comment by nancyt1** [comment]
@tbrlpld Sounds good to me!

**Comment by simont-cr** [comment]
@nancyt1 @kristinashu Is this still something that we will be focusing on this year or are clear to close for the time being?

**Comment by kristinashu** [comment]
I think this is ok to close.
